exporter/otelarrowexporter/internal/arrow/exporter_test.go

```diff
@@ -576,65 +577,92 @@ func TestArrowExporterStreaming(t *testing.T) {
 
-	var expectOutput []metadata.MD
-	var actualOutput []metadata.MD
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
 
-	var wg sync.WaitGroup
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
-		md := metadata.MD{}
-		hpd := hpack.NewDecoder(4096, func(f hpack.HeaderField) {
-			md[f.Name] = append(md[f.Name], f.Value)
-		})
-		for data := range channel.sendChannel() {
-			if len(data.Headers) == 0 {
-				actualOutput = append(actualOutput, nil)
-			} else {
-				_, err := hpd.Write(data.Headers)
+			require.NoError(t, tc.exporter.Start(ctx))
+
+			var expectOutput []metadata.MD
+			var actualOutput []metadata.MD
+
+			var wg sync.WaitGroup
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				md := metadata.MD{}
+				hpd := hpack.NewDecoder(4096, func(f hpack.HeaderField) {
+					md[f.Name] = append(md[f.Name], f.Value)
+				})
+				for data := range channel.sendChannel() {
+					if len(data.Headers) == 0 {
+						actualOutput = append(actualOutput, nil)
+					} else {
+						_, err := hpd.Write(data.Headers)
+						require.NoError(t, err)
+						actualOutput = append(actualOutput, md)
+						md = metadata.MD{}
+					}
+					channel.recv <- statusOKFor(data.BatchId)
+				}
+			}()
+
+			for times := 0; times < 10; times++ {
+				input := testdata.GenerateTraces(2)
+
+				if times%2 == 1 {
+					md := metadata.MD{
+						"expected1":       []string{"metadata1"},
+						"expected2":       []string{fmt.Sprint(times)},
+						"otlp-pdata-size": []string{"329"},
+					}
+					expectOutput = append(expectOutput, md)
+				} else {
+					expectOutput = append(expectOutput, metadata.MD{
+						"otlp-pdata-size": []string{"329"},
+					})
+				}
+
+				sendCtx := ctx
+				if withDeadline {
+					sendCtx, _ = context.WithTimeout(sendCtx, time.Second)
+				}
+
+				sent, err := tc.exporter.SendAndWait(sendCtx, input)
 				require.NoError(t, err)
-				actualOutput = append(actualOutput, md)
-				md = metadata.MD{}
+				require.True(t, sent)
 			}
-			channel.recv <- statusOKFor(data.BatchId)
-		}
-	}()
-
-	for times := 0; times < 10; times++ {
-		input := testdata.GenerateTraces(2)
+			// Stop the test conduit started above.
+			cancel()
+			wg.Wait()
 
-		if times%2 == 1 {
-			md := metadata.MD{
-				"expected1":       []string{"metadata1"},
-				"expected2":       []string{fmt.Sprint(times)},
-				"otlp-pdata-size": []string{"329"},
+			// Manual check for proper deadline propagation.  Since the test
+			// is timed we don't expect an exact match.
+			if withDeadline {
+				for _, out := range actualOutput {
+					dead := out.Get("grpc-timeout")
+					require.Equal(t, 1, len(dead))
+					require.Less(t, 0, len(dead[0]))
+					to, err := grpcutil.DecodeTimeout(dead[0])
+					require.NoError(t, err)
+					// Allow the test to lapse for 0.5s.
+					require.Less(t, time.Second/2, to)
+					require.GreaterOrEqual(t, time.Second, to)
+					out.Delete("grpc-timeout")
+				}
```

Comment:
For the most part, I just re-indented this test. This is all that really changes.